pandas/core/tools/datetimes.py

```diff
@@ -496,6 +499,8 @@ def _to_datetime_with_format(
                 utc = tz == "utc"
                 return _box_as_indexlike(result, utc=utc, name=name)
 
+        # GH#42957 #replacing all na with np.nan
```

Comment:
can't do this as this doesn't allow errors handling. i think the best solution is to indeed do this in cython, but we are going to have to take a dependency on missing.pyx / refactor.

Reply:
Hi @jreback, happy to revert to https://github.com/pandas-dev/pandas/pull/42982/commits/74a8e95bb6ad01ebc83859dadb499f92bbf8386b (fixing in cython with dependency on missing.pyx). 
But the only exception ``array_strptime`` is unable to handle is ``pd.NA``. Other exceptions are handled in ``_array_strptime_with_fallback``. So if we just replace ``pd.NA`` (as and when present) with something ``array_strptime`` is able to handle (like np.nan or None), wouldn't that be ideal?

Reply:
Hi @debnathshoham, I think that it isn't good idea to replace `pd.NA`, cause user (programmer how use `pd.to_datetime`) doesn't want to replace pd.NA. I think that use dependecy from `missing.pyx` is a good idea cause pandas comunity add new type of nan - `pd.NA` and all libs in pandas should use it from core pandas api.

Reply:
The user expectation of `to_datetime` from both np.nan and pd.NA is same (i.e. NaT), right?

Reply:
Yes, but user expect that in his array `pd.NA` still stay` pd.NA` and wouldn't convert to `np.nan`.

Sorry for long time answer, i was busy at this weekend.

Reply:
in the original array `pd.NA` wouldn't be replaced with `np.nan`. It would only be replace when `to_datetime` is called, before calling `_array_strptime_with_fallback` (which can't handle `pd.NA`). Hope the below case will clarify.


```python
>>> import pandas as pd
>>> pd.__version__
'1.4.0.dev0+423.g802e27c437'
>>> a=[pd.NA, 20210815]
>>> pd.to_datetime(a, format="%Y%m%d")
DatetimeIndex(['NaT', '2021-08-15'], dtype='datetime64[ns]', freq=None)
>>> a
[<NA>, 20210815]

```

Reply:
Hi @jreback, could you please check if this looks okay? Otherwise, will be happy to revert to https://github.com/pandas-dev/pandas/commit/74a8e95bb6ad01ebc83859dadb499f92bbf8386b.

Reply:
If we want to treat pd.NA as distinct from NaT, then we wouldnt want to support in in array_strptime, would need a new dtype like #40932.

Reply:
see below you are tesing the errors='ignore' case so hard to tell if this is ok or not.

Reply:
i don't think this solution is good. we now have null handling logic in 2 places. also I dont't believe this actually works correctly  for ignore NA's. please go back to the cython solution.